src/heightmap.cpp

```diff
@@ -22,6 +22,47 @@
 
 #include "safeguards.h"
 
+/**
+ * Maximum number of pixels for one dimension of a heightmap image.
+ *
+ * At some point the number of pixels in a heightmap becomes larger compared to
+ * the pixels that will be taken for the map. With this limit already 75% of the
+ * pixels would be discarded for the largest image and the largest map size.
+ */
```

Comment:
`Be reasonable in how much bigger the heightmap can be versus the biggest map size.`

That alone for me would be sufficient, I think :) Also, I think times two is a bit low .. maybe times 4?

Reply:
Checked all heightmaps on BaNaNaS. Biggest is 8kx8k. So anything equal or above that is fine from that point of view :)